chia/simulator/full_node_simulator.py

```diff
@@ -144,3 +193,250 @@ async def reorg_from_index_to_new_index(self, request: ReorgProtocol):
+            block: FullBlock
+            if guarantee_transaction_blocks:
+                block = await self.farm_new_transaction_block(FarmNewBlockProtocol(farm_to))
+            else:
+                block = await self.farm_new_block(FarmNewBlockProtocol(farm_to))
+            height = uint32(block.height)
+            rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
+
+        peak_height = self.full_node.blockchain.get_peak_height()
+
+        if peak_height is None:
+            raise RuntimeError("Peak height still None after processing at least one block")
+
+        if peak_height < height:
+            raise RuntimeError(f"Peak height lower than expected: {peak_height} < {height}")
+
+        return rewards
+
+    async def farm_blocks_to_wallet(self, count: int, wallet: Wallet) -> int:
+        """Farm the requested number of blocks to the passed wallet. This will
+        process additional blocks as needed to process the reward transactions
+        and also wait for the rewards to be present in the wallet.
+
+        Arguments:
+            count: The number of blocks to farm.
+            wallet: The wallet to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos farmed to the requested address.
+        """
+        if count == 0:
+            return 0
+
+        target_puzzlehash = await wallet.get_new_puzzlehash()
+        rewards = 0
+
+        block_reward_coins = set()
+        expected_reward_coin_count = 2 * count
+
+        # TODO: why 2 and not 1?
+        for index in range(count + 2):
+            if index < count:
+                rewards += await self.process_blocks(
+                    count=1, farm_to=target_puzzlehash, guarantee_transaction_blocks=False
+                )
+            else:
+                await self.process_blocks(count=1, guarantee_transaction_blocks=True)
+
+            peak_height = self.full_node.blockchain.get_peak_height()
+            if peak_height is None:
+                raise RuntimeError("Peak height still None after processing at least one block")
+
+            coin_records = await self.full_node.coin_store.get_coins_added_at_height(height=peak_height)
+            for record in coin_records:
+                if record.coin.puzzle_hash == target_puzzlehash and record.coinbase:
+                    block_reward_coins.add(record.coin)
+
+            if len(block_reward_coins) >= expected_reward_coin_count:
+                break
+        else:
+            raise RuntimeError("Not all reward coins identified")
+
+        if len(block_reward_coins) != expected_reward_coin_count:
+            raise RuntimeError(f"Expected {expected_reward_coin_count} reward coins, got: {len(block_reward_coins)}")
+
+        await wait_for_coins_in_wallet(coins=block_reward_coins, wallet=wallet)
+
+        return rewards
+
+    async def farm_rewards_to_wallet(self, amount: int, wallet: Wallet) -> int:
```

Comment:
This function seems redundant and almost the same as the previous one

Reply:
It is part of trying to work away from tests being coded based around inner workings of the blockchain.  A lot of tests are not about farming yet they have farming loops where they depend on the reward size instead of just trying to make sure they have some amount of xch/coins available so they can run.  I don't think this is the final form but I do think a majority of tests probably shouldn't have any code about farming rewards or the number of blocks needed to get enough coins.